test/integration/query-interface/schemas.test.ts

```diff
@@ -3,37 +3,91 @@ import { spy } from 'sinon';
 import type { CreateSchemaQueryOptions } from '@sequelize/core/_non-semver-use-at-your-own-risk_/dialects/abstract/query-generator';
 import { sequelize } from '../support';
 
-const queryInterface = sequelize.queryInterface;
-
+const { queryInterface, dialect } = sequelize;
 const testSchema = 'testSchema';
 
-describe('QueryInterface#createSchema', async () => {
-  if (!sequelize.dialect.supports.schemas) {
-    return;
-  }
+// MySQL and MariaDB view databases and schemas as identical. Other databases consider them separate entities.
+const dialectsWithEqualDBsSchemas = ['mysql', 'mariadb'];
+
+describe('QueryInterface#{create,drop,dropAll,showAll}Schema', () => {
+  if (dialect.supports.schemas) {
```

Comment:
To reduce nesting I would still use `if (!dialect.supports.schemas)` and then run that and return so we do not need to use else